src/sentry/integrations/github/client.py

```diff
@@ -18,6 +18,11 @@
 
 logger = logging.getLogger("sentry.integrations.github")
 
+# Some functions that require a large number of API requests can use this value
+# as the lower ceiling before hitting Github anymore, thus, leaving at least these
+# many requests left for other features that need to reach Github
+MINIMUM_REQUESTS = 200
```

Comment:
How did we arrive at 200 being the appropriate value here?

Reply:
There's no math to support it. I may need to raise it if we still see it.

The logging for the errors will include the rate before we exhaust it, thus, being able to learn.
<img width="425" alt="image" src="https://user-images.githubusercontent.com/44410/217052433-cb39ad60-e1b9-4172-b1ca-6dc624b8317e.png">


If we get too close to 0, we may still hit API rate exhaustion since other features may still consume 200 requests within an hour. I know that at least code derivation should not be exhaust it.

Reply:
Good question!